homeassistant/components/simplisafe/__init__.py

```diff
@@ -41,53 +40,55 @@
 
 DATA_LISTENER = "listener"
 
-ATTR_ARMED_LIGHT_STATE = "armed_light_state"
-ATTR_ARRIVAL_STATE = "arrival_state"
+ATTR_ALARM_DURATION = "alarm_duration"
+ATTR_ALARM_VOLUME = "alarm_volume"
+ATTR_CHIME_VOLUME = "chime_volume"
+ATTR_ENTRY_DELAY_AWAY = "entry_delay_away"
+ATTR_ENTRY_DELAY_HOME = "entry_delay_home"
+ATTR_EXIT_DELAY_AWAY = "exit_delay_away"
+ATTR_EXIT_DELAY_HOME = "exit_delay_home"
+ATTR_LIGHT = "light"
 ATTR_PIN_LABEL = "label"
 ATTR_PIN_LABEL_OR_VALUE = "label_or_pin"
 ATTR_PIN_VALUE = "pin"
-ATTR_SECONDS = "seconds"
 ATTR_SYSTEM_ID = "system_id"
-ATTR_TRANSITION = "transition"
-ATTR_VOLUME = "volume"
-ATTR_VOLUME_PROPERTY = "volume_property"
+ATTR_VOICE_PROMPT_VOLUME = "voice_prompt_volume"
 
-STATE_AWAY = "away"
-STATE_ENTRY = "entry"
-STATE_EXIT = "exit"
-
-VOLUME_PROPERTY_ALARM = "alarm"
-VOLUME_PROPERTY_CHIME = "chime"
-VOLUME_PROPERTY_VOICE_PROMPT = "voice_prompt"
+VOLUMES = [VOLUME_OFF, VOLUME_LOW, VOLUME_MEDIUM, VOLUME_HIGH]
 
 SERVICE_BASE_SCHEMA = vol.Schema({vol.Required(ATTR_SYSTEM_ID): cv.positive_int})
 
 SERVICE_REMOVE_PIN_SCHEMA = SERVICE_BASE_SCHEMA.extend(
     {vol.Required(ATTR_PIN_LABEL_OR_VALUE): cv.string}
 )
 
-SERVICE_SET_DELAY_SCHEMA = SERVICE_BASE_SCHEMA.extend(
-    {
-        vol.Required(ATTR_ARRIVAL_STATE): vol.In((STATE_AWAY, STATE_HOME)),
-        vol.Required(ATTR_TRANSITION): vol.In((STATE_ENTRY, STATE_EXIT)),
-        vol.Required(ATTR_SECONDS): cv.positive_int,
-    }
-)
-
-SERVICE_SET_LIGHT_SCHEMA = SERVICE_BASE_SCHEMA.extend(
-    {vol.Required(ATTR_ARMED_LIGHT_STATE): cv.boolean}
-)
-
 SERVICE_SET_PIN_SCHEMA = SERVICE_BASE_SCHEMA.extend(
     {vol.Required(ATTR_PIN_LABEL): cv.string, vol.Required(ATTR_PIN_VALUE): cv.string}
 )
 
-SERVICE_SET_VOLUME_SCHEMA = SERVICE_BASE_SCHEMA.extend(
+SERVICE_SET_SYSTEM_PROPERTIES_SCHEMA = SERVICE_BASE_SCHEMA.extend(
     {
-        vol.Required(ATTR_VOLUME_PROPERTY): vol.In(
-            (VOLUME_PROPERTY_ALARM, VOLUME_PROPERTY_CHIME, VOLUME_PROPERTY_VOICE_PROMPT)
+        vol.Optional(ATTR_ALARM_DURATION): vol.All(
+            vol.Coerce(int), vol.Range(min=30, max=480)
```

Comment:
For longer time periods than a few seconds we normally use `cv.time_period` as input validator. If the api requires integer seconds, convert the period to that at the end of the validation.

Reply:
I chose straight integers because they could have min/max limits applied easily; I don't see any comparable mechanism for `cv.time_period` – thoughts?

Reply:
Convert the time period to seconds and apply the limits as before.

Reply:
Sorry, not following – I understand that `cv.time_period` spits out a `timedelta`, but can the seconds conversion occur _within_ the schema validation?

Reply:
Is this what lambdas are used for in `voluptuous`?

```
vol.Optional(ATTR_ALARM_DURATION): vol.All(
    cv.time_period, lambda value: value.seconds, vol.Range(min=30, max=480)
)
```

Reply:
Yeah, that looks good! :+1: